src/protocol/basics/mul/mod.rs

```diff
@@ -51,33 +51,33 @@ use {malicious::multiply as malicious_mul, semi_honest::multiply as semi_honest_
 
 /// Implement secure multiplication for semi-honest contexts with replicated secret sharing.
 #[async_trait]
-impl<'a, F: Field> SecureMul<SemiHonestContext<'a>> for Replicated<F> {
+impl<C: Context, F: Field> SecureMul<C> for Replicated<F> {
```

Comment:
Ah, this was a particularly challenging part of the change.  So it's probably worth explaining.

We have four contexts: `SemiHonest`, `Malicious`, and upgraded versions of the two.  This implementation needs to hit three of those four, which all use `Replicated<F>` (`UpgradedMaliciousContext` uses `MaliciousReplicated<F>`).  This could be set to bound `C` to `UpgradableContext`, but then I'd need to add another implementation for `UpgradedSemiHonestContext` as well.  This was easier.